resources/services/backup/vaults.go

```diff
@@ -0,0 +1,389 @@
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	result, err := svc.GetBackupVaultAccessPolicy(
+		ctx,
+		&backup.GetBackupVaultAccessPolicyInput{BackupVaultName: vault.BackupVaultName},
+		func(o *backup.Options) { o.Region = cl.Region },
+	)
+	if err != nil {
+		if cl.IsNotFoundError(err) {
+			return nil
+		}
+		return diag.WrapError(err)
+	}
+	return diag.WrapError(resource.Set(c.Name, result.Policy))
+}
+
+func resolveVaultNotifications(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource) error {
+	vault := resource.Item.(types.BackupVaultListMember)
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	result, err := svc.GetBackupVaultNotifications(
+		ctx,
+		&backup.GetBackupVaultNotificationsInput{BackupVaultName: vault.BackupVaultName},
+		func(o *backup.Options) { o.Region = cl.Region },
+	)
+	if err != nil {
+		var ae smithy.APIError
+		if !errors.As(err, &ae) {
+			return diag.WrapError(err)
+		}
+		if ae.ErrorCode() == "ERROR_2106" {
+			// trying to ignore "ERROR_2106: Failed reading notifications from database for Backup vault ..."
+			return nil
+		}
+		return diag.WrapError(err)
+	}
+	if err := resource.Set("notification_events", result.BackupVaultEvents); err != nil {
+		return diag.WrapError(err)
+	}
+	return diag.WrapError(resource.Set("notification_sns_topic_arn", result.SNSTopicArn))
+}
+
+func fetchVaultRecoveryPoints(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	vault := parent.Item.(types.BackupVaultListMember)
+	params := backup.ListRecoveryPointsByBackupVaultInput{BackupVaultName: vault.BackupVaultName, MaxResults: aws.Int32(100)}
+	for {
+		result, err := svc.ListRecoveryPointsByBackupVault(ctx, &params, func(o *backup.Options) {
+			o.Region = cl.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		res <- result.RecoveryPoints
+		if aws.ToString(result.NextToken) == "" {
+			break
+		}
+		params.NextToken = result.NextToken
+	}
+	return nil
+}
+
+func resolveVaultRecoveryPointCreatedBy(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	rp := resource.Item.(types.RecoveryPointByBackupVault)
+	b, err := json.Marshal(rp.CreatedBy)
+	if err != nil {
+		return diag.WrapError(err)
+	}
+	return diag.WrapError(resource.Set(c.Name, b))
+}
+
+func resolveRecoveryPointTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	rp := resource.Item.(types.RecoveryPointByBackupVault)
+	cl := meta.(*client.Client)
+	svc := cl.Services().Backup
+	params := backup.ListTagsInput{ResourceArn: rp.RecoveryPointArn}
+	tags := make(map[string]string)
+	for {
+		result, err := svc.ListTags(ctx, &params, func(o *backup.Options) { o.Region = cl.Region })
+		if result == nil {
+			break
+		}
+		if err != nil {
+			return diag.WrapError(err)
+		}
```

Comment:
Shouldn't this be switched? So we check the error first?

Reply:
Sure, my bad.